### ISSUE TITLE
Support in browser and Node environments. 

### ISSUE THREAD

**Issue by darioAnongba** [rust-bitcoin/corepc#34]
Hi everyone, 

I arrived in this repo by doing a deep dive starting from the BitcoinDevKit. I am writing a library called [`bdk-wasm`](https://github.com/MetaMask/bdk-wasm) enabling usage of BDK in the browser and Node by compiling it to WASM and creating TypeScript bindings. 

I was surprised when I saw that BDK, when using the `bitcoind_rpc` feature, compiles to WASM just fine and can be executed. It sadly fails in the browser and Node with the following error: 
```
JsonRpc(Transport(SocketError(Error { kind: Unsupported, message: \"operation not supported on this platform\" })))
```

The Bitcoin node exposes over HTTP and we can successfully query `Esplora` from the browser and Node using [`esplora_client`](https://github.com/bitcoindevkit/rust-esplora-client). This package uses [`reqwest`](https://github.com/seanmonstar/reqwest) under the hood that has a target `"wasm32-unknown-unknown"`. Meaning that I think this package could work easily within browser and Node when compiled to WASM.

## Proposal
Add a `feature` or `target` mechanism that uses `reqwest`, adding support for Browser and Node environments.

Let me know if you think this won't work or if it's a bad idea. I'm happy to implement the changes myself.

**Comment by darioAnongba** [comment]
Hahaha understandable. Let's keep this open and see as it's true that we shouldn't expect someone to maintain this if there is no strong demand for it.

**Comment by stevenroose** [comment]
Just indicating interest in having bitreq work in WASM.

**Comment by tnull** [comment]
> Just indicating interest in having bitreq work in WASM.

Yes, we might soon look into adding WASM support, as we'll need it on the LDK side, too.

**Comment by jamillambert** [comment]
Issue moved to Forgejo: https://gitea.bitcoin.ninja/rust-bitcoin/corepc/issues/34
